### PR TITLE
Remove legacy query from parameter targets

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/38650

I tried maybe 20 different approaches to migrate this check from legacy query. The most important issue (for future reference) is that we have parameters with field targets in embedding where the query is not available (it's either not present at all or just `{ "type:": "query" }` without even `source-table`, so existing MLv2 functions cannot be used. 

I've came up with a very simple metadata-based retrieval that works for **parameters** that involves parsing the field ref in MLv1. This is based on the fact that parameters "care" only for real, database-based fields (either coming from the table or model metadata overrides), and not arbitrary columns.

### CODE REVIEW

File: frontend/src/metabase/selectors/metadata.ts [hunk 1]
```diff
@@ -131,7 +131,9 @@ export const getMetadata: (
       Object.values(tables).map(t => [t.id, createTable(t, metadata)]),
     );
     metadata.fields = Object.fromEntries(
-      Object.values(fields).map(f => [f.uniqueId, createField(f, metadata)]),
+      Object.values(fields)
+        .filter(f => f.uniqueId != null)
```
Comment (ranquild) [inline frontend/src/metabase/selectors/metadata.ts:135]: Remove broken field entries created for field values without any field info

File: frontend/src/metabase-lib/parameters/utils/cards.ts [hunk 2]
```diff
@@ -37,6 +38,9 @@ export function getCardUiParameters(
       };
     }
 
-    return { ...parameter, hasVariableTemplateTagTarget: true };
+    return {
+      ...parameter,
+      hasVariableTemplateTagTarget: !isDimensionTarget(target),
```
Comment (ranquild) [inline frontend/src/metabase-lib/parameters/utils/cards.ts:43]: Now `getParameterTargetField` doesn't return broken/stub `Field` instances. Previously:
- broken `Field` instances were always returned by `ExpressionDimension`
- broken `Field` instances could be returned by `FieldDimension` in some cases when there was no match by field ref

We now don't create broken `Field` instances. With this change, if `field == null`, it no longer means that the parameter is mapped to a text variable (e.g. no field filter), so we need to add this check here.

File: frontend/src/metabase-lib/parameters/utils/targets.ts [hunk 3]
```diff
@@ -41,16 +40,30 @@ export function getParameterTargetField(
...
+  if (isTemplateTagReference(fieldRef)) {
+    const dimension = TemplateTagDimension.parseMBQL(
+      fieldRef,
+      metadata,
+      question.legacyQuery() as NativeQuery,
+    );
     return dimension?.field();
   }
 
+  if (isConcreteFieldReference(fieldRef)) {
```
Comment (ranquild) [inline frontend/src/metabase-lib/parameters/utils/targets.ts:61]: We are still in the MLv1 land, not entirely MLv2 based. But we no longer parse the query. The check is always far more stable and predictable than before.

File: frontend/src/metabase/dashboard/selectors.unit.spec.js [hunk 4]
```diff
@@ -55,20 +60,11 @@ const STATE = {
...
-    tables: {},
-    fields: {
-      1: { id: 1 },
-      2: { id: 2 },
-    },
-    metrics: {},
-    segments: {},
-    questions: {},
-  },
+  entities: createMockEntitiesState({
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/selectors.unit.spec.js:63]: Create real, not broken field instances

**Review by uladzimirdev — APPROVED** [review]
LGTM

**Pull request merged**